.github/workflows/stable.yml

```
@@ -27,59 +53,144 @@ jobs:
+    # that were built by docker buildx (the build-push-action step) locally. They get pushed to dockerhub but are
+    # only identifiable by their digest and it appears docker does not let us select an image that way.
+    # Not the most elegant, but it works.
+    - name: Store tag info
+      if: ${{ steps.stablecheck.outputs.BUILD }}
+      shell: bash
+      run: |
+        mkdir -p /tmp/tags
         RUST_DATE="$(date +"%Y-%m-%d")"
         RUST_CHANNEL=stable
-        echo "TAG1=clux/muslrust:${RUST_CHANNEL}" >> $GITHUB_ENV
-        echo "TAG2=clux/muslrust:${RUST_VER}-${RUST_CHANNEL}" >> $GITHUB_ENV
-        echo "TAG3=clux/muslrust:${RUST_VER}" >> $GITHUB_ENV
-      if: ${{ steps.stablecheck.outputs.BUILD }}
+        RUST_VER="$(docker run --platform ${{ matrix.platform }} rustmusl-temp rustc --version | grep -oE "[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]")"
+
+        echo $RUST_DATE > /tmp/tags/rust-date
+        echo $RUST_CHANNEL > /tmp/tags/rust-channel
+        echo $RUST_VER > /tmp/tags/rust-ver
 
-    - name: Echo tags
+    - name: Tag and push
+      if: ${{ steps.stablecheck.outputs.BUILD }}
       shell: bash
       run: |
-        echo $TAG1
-        echo $TAG2
-        echo $TAG3
+        RUST_DATE=$(cat /tmp/tags/rust-date)
+        RUST_CHANNEL=$(cat /tmp/tags/rust-channel)
+        RUST_VER=$(cat /tmp/tags/rust-ver)
 
-    - name: Run tests
+        TAG_NAME="${{ matrix.arch }}-${RUST_VER}-${RUST_CHANNEL}-${RUST_DATE}"
+
+        docker tag rustmusl-temp ${{ env.REGISTRY_IMAGE }}:$TAG_NAME
+        docker push ${{ env.REGISTRY_IMAGE }}:$TAG_NAME
+
+    - name: Upload tags
+      if: ${{ steps.stablecheck.outputs.BUILD }}
+      uses: actions/upload-artifact@v4
+      with:
+        name: tags
+        path: /tmp/tags
+        if-no-files-found: error
+        retention-days: 1
+        overwrite: true
+
+  merge:
+    name: 'Stable merge'
+    runs-on: ubuntu-latest
+    needs:
+      - build
+    steps:
+    - uses: 'actions/checkout@v2'
+    - name: Check if we need a new stable
+      id: stablecheck
       shell: bash
+      run: |
+        pip3 install --user toml
+        if python3 check_stable.py; then
+          echo "Stable tag missing; running all build steps"
+          echo '::set-output name=BUILD::YES'
+        else
+          echo "Stable tag found; skipping all build steps"
+        fi
+
+    - name: Download tags
       if: ${{ steps.stablecheck.outputs.BUILD }}
-      run: just test
+      uses: actions/download-artifact@v4
+      with:
+        path: /tmp/tags
+        name: tags
 
-    - name: Login to DockerHub
-      if: github.event_name != 'pull_request'
-      uses: docker/login-action@v1
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        images: ${{ env.REGISTRY_IMAGE }}
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
```

```suggestion
    - name: Login to Docker Hub
      uses: docker/login-action@v3
      if: false
```
maybe put in a safety thing here while we get this to build